app/types/index.ts

```diff
@@ -1,52 +1,53 @@
 export interface POPUP_PROPS {
-  tabId : number
-  tabUrl : string
-  enabled : boolean
-  requests : Array<any>
-  errorMessage : string
-  checkedReqs : object
-  statusCodes: object
-  responseText: object
-  contentType: object
-  PageDetails : object
-  interceptStatus : string
-  isInterceptorOn: interceptOn
+  tabId: number;
+  tabUrl: string;
+  enabled: boolean;
+  method?: string;
+  url?: string;
+  requests: Array<any>;
+  errorMessage: string;
+  checkedReqs: object;
+  statusCodes: object;
+  responseText: object;
+  contentType: object;
+  PageDetails: Array<object>;
+  interceptStatus: string;
```

Comment:
instead of `interceptSatus` as `string`. it can be union/enum state. 
https://www.typescriptlang.org/docs/handbook/enums.html